src/triggers/messages.ts

```diff
@@ -0,0 +1,51 @@
+import { BigNumber } from 'ethers';
+import _ from 'lodash';
+
+import { Table } from '../db/db';
+import { CrdtMessage } from '../types/message';
+import { Trigger } from '../types/trigger';
+
+
+export const newMessages: Trigger<undefined> = async (clients) => {
+  // get all unprocessed messages
+  const selectSql = `select m.* from ${Table.crdtMessages} m
```

Comment:
This will only fetch `QUERY_TRIGGER_BATCH_SIZE` rows, so if there are more than `QUERY_TRIGGER_BATCH_SIZE` stale messages the trigger won't actually process properly. Currently as we are the only one creating messages, all messages should be written to processed_messages so this shouldn't be a problem immediately, but will be in the future once we start syncing changes from other nodes. Maybe this trigger should also pass a list of stale messages to the processor, so that the processor can skip processing them and just write them directly to processed messages so that the trigger can ignore them

Reply:
yeah, i think stale messages should still be processed (and just do nothing as part of their processing)

Reply:
the existing newMessages trigger i suppose is something like a "restore-from-backup" solution, where the intention is for the table to be append-only and for every message to be processed.

i imagine when we have other nodes and peer with them, we'll have a separate table and trigger for those, like a mempool, where stale messages are just dropped/deleted from the table